lib/collection/src/shards/replica_set/update.rs

```diff
@@ -220,6 +224,33 @@ impl ShardReplicaSet {
 
         let (successes, failures): (Vec<_>, Vec<_>) = all_res.into_iter().partition_result();
 
+        // Advance clock if some replica echoed *newer* tick
+
+        let new_clock_tick = successes
+            .iter()
+            .filter_map(|(_, result)| {
+                let echo_tag = result.clock_tag?;
+
+                if echo_tag.peer_id != clock_tag.peer_id {
+                    // TODO: `log::warn`!?
+                    return None;
+                }
```

Comment:
Is there a good reason to explicitly reject this?

Reply:
It's more of a future-proofing/fail-safe.

With replication/shard proxies/write ordering it is sometimes very unobvious which code path a request will take, and it seems possible to me, that we might introduce a bug that will assign the wrong clock tag to the `UpdateResponse`?